features/admin.actions.v1/components/action-config-form.tsx

```diff
@@ -0,0 +1,829 @@
+                        + ".description"),
+                    level: AlertLevels.ERROR,
+                    message: t("console:manage.features.actions.notification.genericError." + operation
+                        + ".message")
+                })
+            );
+        }
+    };
+
+    const getFieldDisabledStatus = (): boolean => {
+        if (isCreateFormState) {
+            return !hasActionCreatePermissions;
+        } else {
+            return !hasActionUpdatePermissions;
+        }
+    };
+
+    const validateForm = (values: ActionConfigFormPropertyInterface): ActionConfigFormPropertyInterface => {
+        const error: ActionConfigFormPropertyInterface = {
+            accessTokenAuthProperty: undefined,
+            authenticationType: undefined,
+            endpointUri: undefined,
+            headerAuthProperty: undefined,
+            id: undefined,
+            name: undefined,
+            passwordAuthProperty: undefined,
+            usernameAuthProperty: undefined,
+            valueAuthProperty: undefined
+        };
+
+        if (!values?.name) {
+            error.name = t("console:manage.features.actions.fields.name.validations.empty");
+        }
+        const vsCharRegex: RegExp = /^[A-Za-z0-9 -_]*$/;
+
+        if (!vsCharRegex.test(values?.name)) {
+            error.name = t("console:manage.features.actions.fields.name.validations.invalid");
+        }
+        if (!values?.endpointUri) {
+            error.endpointUri = t("console:manage.features.actions.fields.endpoint.validations.empty");
+        }
+        if (URLUtils.isURLValid(values?.endpointUri)) {
+            if (!(URLUtils.isHttpsUrl(values?.endpointUri))) {
+                error.endpointUri = t("console:manage.features.actions.fields.endpoint.validations.notHttps");
+            }
+        } else {
+            error.endpointUri = t("console:manage.features.actions.fields.endpoint.validations.invalidUrl");
+        }
+        if (!values?.authenticationType) {
+            error.authenticationType = t("console:manage.features.actions.fields.authenticationType.validations.empty");
+        }
+
+        switch (authenticationType) {
+            case AuthenticationType.BASIC:
+                if(isCreateFormState || isAuthenticationUpdateFormState ||
+                    values?.usernameAuthProperty || values?.passwordAuthProperty) {
+                    if (!values?.usernameAuthProperty) {
+                        error.usernameAuthProperty = t("console:manage.features.actions.fields.authentication." +
+                            "types.basic.properties.username.validations.empty");
+                    }
+                    if (!values?.passwordAuthProperty) {
+                        error.passwordAuthProperty = t("console:manage.features.actions.fields.authentication." +
+                            "types.basic.properties.password.validations.empty");
+                    }
+                }
```

Comment:
It needs to be checked because, we need to validate whether the user has added values for both fields. Currently we are not supporting to update one property